google_guest_agent/events/sshtrustedca/sshtrustedca_linux.go

```diff
@@ -48,13 +49,30 @@ func createNamedPipe(pipePath string) error {
 	return nil
 }
 
+// finishedCb is used by the event handler to communicate the write to the
+// pipe is finised.
+func (mp *Watcher) finishedCb() {
+	mp.waitingWrite = false
+}
+
 // Run listens to ssh_trusted_ca's pipe open calls and report back the event.
 func (mp *Watcher) Run(ctx context.Context) (bool, string, interface{}, error) {
 	var canceled bool
 
+	for mp.waitingWrite {
```

Comment:
If we're expecting to use `waitingWrite` in multiple contexts, we should wrap this in a getter / setter that uses a mutex to protect reads / writes.

Reply:
seems like a good call. will do.

Reply:
Done. Thanks.